csrc/scheduler/pointwise.cpp

```diff
@@ -484,8 +487,11 @@ std::unique_ptr<PointwiseParams> getPointwiseHeuristics(
     }
   }
 
+  // If we have sub-byte data types, we wouldn't want to clamp vectorization
+  // factor to 1, otherwise we could end up with illegal array type with
+  // sub-byte length.
   params->vectorization_factor = std::min(
-      max_vect_factor,
+      has_sub_byte ? std::max(2l, max_vect_factor) : max_vect_factor,
```

Comment:
Thanks - I need something like this as well.  I thought I may need this to have a minimum of 4/8(bf16).

Do you think this may cause problems with alignment? If we can't vectorize and we still set this to 2/4/8? I am not sure what the solution to that case would be.

Reply:
Do you think we should assert on the params->vectorization_factor here in the case that we have a packed fp4 type and we aren't able to vectorize to 2 or more?

Reply:
^ I would prefer not.

generally speaking, earlier error seems like a good idea. But vectorization should not be the answer.
It's really the inline that's failing us. i.e. we don't necessarily need the fp4 TV to be vectorized if we have correct inline. We really only needed to have a packed storage.

I don't know how to fix that properly yet and I wanted it to work for now. That's why I was calling it a hacky patch. 😰

Reply:
Sorry, I don't understand what you meant by "the inline".